### PR TITLE
Visual Studio 2017 solution, v2 !

### PR DESCRIPTION
**Pull request by aybe**
Lesson learned, thanks :)

I have started from scratch ...

**could you use such a list globally instead, as in git config core.excludeFiles ~/.gitignore_global?**
I've reduced ignores to the [strict minimum](https://github.com/aybe/aubio/commit/0be9abdca1f40b806aad403eaeb2ba342d3070ad)

**why are the source files binaries?**
I don't know why they were, they shouldn't and [they aren't now](https://github.com/aybe/aubio/commit/881d4586e8439fad05c2b8c08cde6b797fd60102)
(hint: I suspect Team Explorer with git client, it is riddled with bugs even though it is tightly integrated in VS)

**what does MSVC say about those `void*` cast?**
My bad, I shouldn't have rushed ...

**AUBIO_API: there must be a better way really.**
Ditto ...

**Did you try building 0.4.6 with waf?**
It used to work before I switched to Win10+VS2017, sorry but I don't remember the exact error messages, might give it another try in the future
PS and my objectives were to be able to step-through aubio in Visual Studio

**Now as mentioned ealier, maintaining these files would be hard for us since it's a tool we don't really use.**
I will maintain the VS solution, it's only a matter of dragging and dropping files onto the solution since this new PR is much less intrusive than my previous one.
And compared to [my old contribution](https://github.com/aubio/aubio/pull/25#issuecomment-285339962), the .NET version of aubio now [sits in its own repository](https://github.com/aybe/aubio.net) (much better)

**Notes**

Life would be much easier if aubio supported CMake, I somehow tried to put [the aubio CMakeLists.txt from vpckg](https://github.com/Microsoft/vcpkg/blob/master/ports/aubio/CMakeLists.txt) but that didn't work ... There is a strong possibility that something very simple is happening but since I'm a 99% C# coder, I don't have the necessary knowledge.

Forgot to mention, I rolled out ```AUBIO_API``` because, ironically I didn't find a way to export methods that weren't marked with ```__declspec(dllexport)``` without a ```.lib``` file and didn't want to type all these exports by hand. That file was not being generated previously, now it is so I guess problem is solved. CMake has that magic ```CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS``` without ```__declspec(dllexport)``` but since my attempt to use it failed, I went another, uncomfortable way. 

---

Now, I'm crossing my fingers for the build to pass 🥇

### CODE REVIEW

**Comment by aybe** [comment]
Looks like it's going to pass, I'll take at look at CMake to see if I can make a CMakeLists.txt ...

**Comment by aybe** [comment]
Well ... I've tried a few approaches and to say the least, I don't know what to do now with this PR:

 - https://github.com/Microsoft/vcpkg has aubio and it works out of the box, though UWP flavor fails to build because they use libflac which uses calls not allowed in UWP. Furthermore, this does not give me access to a solution I can eventually commit to.

 - I've been able to tweak the CMakeLists from vcpkg to run with this repo, this approach generates VS solutions, even though there are two being generated (x86/x64). But since vcpkg completely integrates a package (all you need is an `#include`), it kinds of gets obsoleted by solution 1.

 - And there's my VS solution, which works even though it's been done manually. Ultimately I'll keep it since it allows me to browse aubio in a friendly manner. Personally I'm even questioning about the usefulness of my .NET wrapper since I can use C++ within UWP; the strong argument being the much longer development time and headaches in C++ :)

I'll just close this PR for now, use aubio for my own app and see how the situation unwinds with vcpkg.

Thanks for your time !

**Comment by bagong** [comment]
@aybe , referring to solution 1: I am not sure what you mean by "not a solution I can eventually commit to". If it's about the accessibility of vcpkg ports, I can say from experience that MS is extremely welcoming and helpful to contributors to their ports. VCPG is a strongly user driven project where MS mainly plays the role of a helpful moderator. I am sure they would be interested in an update to 0.46 and even more so to work on the UWP build. I was planning to look into updating the port to 0.46, just haven't come around yet. If you'd go a head before me (or anybody else) that would be highly welcome!

**Comment by aybe** [comment]
I was talking about being able to use the solution from within VS, with Intellisense, Resharper and so on. 

Spent a day with vcpkg, it's pretty cool but does not generate friendly VS solutions, you only have the ability to open a folder as a solution, but since their system is complex, it's not picking the paths from CMake and so on (yet), therefore it's just like I'm using grep and notepad.

Good news though, I've patched libflac and libsndfile for aubio on UWP for vcpkg, but aubio too needs some patching as well for UWP...

long story short : the VS solution in the end is good as you are a first-class customer, I will consider another PR if I can get a generic CMakeLists up for aubio !

**Comment by aybe** [comment]
To let you know, I've submitted a patch for aubio that builds for UWP:

https://github.com/Microsoft/vcpkg/pull/2216

I can finally start coding now :)
